app/models/internal/impl/DefaultVersionSpecificationAttribute.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.internal.impl;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.logback.annotations.Loggable;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import models.internal.VersionSpecificationAttribute;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+
+/**
+ * Default internal model implementation for a version specification attribute.
+ *
+ * @author Matthew Hayter (mhayter at groupon dot com)
+ */
+@Loggable
+public final class DefaultVersionSpecificationAttribute implements VersionSpecificationAttribute {
+    @Override
+    public String getKey() {
+        return _key;
+    }
+
+    @Override
+    public String getValue() {
+        return _value;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(final Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (!(other instanceof DefaultVersionSpecificationAttribute)) {
+            return false;
+        }
+
+        final DefaultVersionSpecificationAttribute otherDefaultVersionSpecificationAttribute = (DefaultVersionSpecificationAttribute) other;
+        return Objects.equal(_key, otherDefaultVersionSpecificationAttribute._key)
+                && Objects.equal(_value, otherDefaultVersionSpecificationAttribute._value);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(_key, _value);
+    }
+
+    @Override
```

Comment:
Same note about the override documentation. Please make sure it is applied everywhere.